### PR TITLE
Ability to specify multiple paths for migrations and seeds

### PR DESCRIPTION
**Pull request by bartosz-maciaszek**
This PR introduces an ability to specify multiple paths for migrations and seeds. Since [the other pull request that addressed it](https://github.com/robmorgan/phinx/pull/742) had been abandoned, I decided to use previously implemented feature and create a fresh pull request, branching off from the latest `0.5.x-dev`.

Additionally allowed using curly braces in seed paths with GLOB_BRACE (if available). Phenix already uses this feature for migration paths, so I think it's natural to have it for seed paths as well.

### CODE REVIEW

**Comment by bartosz-maciaszek** [comment]
I have no idea why appveyor build fails. It complains about missing migration file when checked with `$this->assertFileExists()`, however `scandir()` indicates the file exists in the directory.

When I removed that assertion completely (to see how `file_get_contents()` handles it), appveyor said:

`file_get_contents(C:\Users\appveyor\AppData\Local\Temp\1\migrations\20160620102959_simple3.php): failed to open stream: Invalid argument`

Did I miss something? All tests pass on travis and locally...

**Comment by rquadling** [comment]
If you reinstate `\Test\Phinx\Console\Command\CreateTest::testSimpleTemplateGeneratorsIsCorrectlyPopulated()`, do things work correctly?

This test is making sure that the template substitution is all working correctly for when a migration is generated by a template. I added this on Friday.

**Comment by bartosz-maciaszek** [comment]
No, it failed locally due to the fact `src/Phinx/Console/Command/AbstractCommand.php` has been modified. That's why I altered it in the first place.

Has it ever passed in the appveyor after your changes?

**Comment by rquadling** [comment]
It is currently building fine! https://ci.appveyor.com/project/robmorgan/phinx/build/1.0.1204

**Comment by bartosz-maciaszek** [comment]
Ok, let me try again.

**Comment by rquadling** [comment]
I see the changes made to AbstractCommand that report what migration paths are being used.

That has upset the expected output of the create command.

Maybe changing ...

```php
preg_match('`^using migration path (.*?)\s+using migration base class (.*?)\s+using template creation class (.*?)\s+created (\1.(\d++)(.*?))\s+`', $commandTester->getDisplay(), $match);
```
to
```
preg_match('`using migration base class (.*?)\s+using template creation class (.*?)\s+created (.*?(\d{14})(.*?))\s+`', $commandTester->getDisplay(), $match);
```
with adjustments to the match offsets, or introduce named captures
```
        // Get output.
        preg_match('`using migration base class (?P<MigrationBaseClass>.*?)\s+using template creation class (?P<TemplateCreationClass>.*?)\s+created (?P<MigrationFilename>.*?(?P<Version>\d{14})(.*?))\s+`', $commandTester->getDisplay(), $match);

        // Was migration created?
        $this->assertFileExists($match['MigrationFilename'], 'Failed to create migration file from template generator');

        // Get migration.
        $actualMigration = file_get_contents($match['MigrationFilename']);

        // Does the migration match our expectation?
        $expectedMigration = "useClassName Phinx\\Migration\\AbstractMigration / className {$commandLine['name']} / version {$match['Version']} / baseClassName AbstractMigration";
        $this->assertSame($expectedMigration, $actualMigration, 'Failed to create migration file from template generator correctly.');
```

**Comment by bartosz-maciaszek** [comment]
Hmm, that's interesting. At the moment it passes on both travis and appveyor but fails locally.

Let me try with the code you've just provided.

**Comment by rquadling** [comment]
Even simpler extraction.

```php
// Get output.
preg_match('`created (?P<MigrationFilename>.+(?P<Version>\d{14}).*?)\s`', $commandTester->getDisplay(), $match);

// Was migration created?
$this->assertFileExists($match['MigrationFilename'], 'Failed to create migration file from template generator');

// Get migration.
$actualMigration = file_get_contents($match['MigrationFilename']);

// Does the migration match our expectation?
$expectedMigration = "useClassName Phinx\\Migration\\AbstractMigration / className {$commandLine['name']} / version {$match['Version']} / baseClassName AbstractMigration";
$this->assertSame($expectedMigration, $actualMigration, 'Failed to create migration file from template generator correctly.');
```

File: tests/Phinx/Console/Command/CreateTest.php [hunk 1]
```diff
@@ -328,16 +328,16 @@ public function testSimpleTemplateGeneratorsIsCorrectlyPopulated(array $config,
         $commandTester->execute($commandLine, array('decorated' => false));
 
         // Get output.
-        preg_match('`^using migration paths\s+ \- (.*?)\s+using migration base class (.*?)\s+using template creation class (.*?)\s+created (\1.(\d++)(.*?))\s+`', $commandTester->getDisplay(), $match);
+        preg_match('`using migration base class (?P<MigrationBaseClass>.*?)\s+using template creation class (?P<TemplateCreationClass>.*?)\s+created (?P<MigrationFilename>.*?(?P<Version>\d{14})(.*?))\s+`', $commandTester->getDisplay(), $match);
```
Comment (rquadling) [inline tests/Phinx/Console/Command/CreateTest.php]: Really only need ```preg_match('`created (?P<MigrationFilename>.+(?P<Version>\d{14}).*?)\s`', $commandTester->getDisplay(), $match);```

**Comment by bartosz-maciaszek** [comment]
Looks like this is it :)

**Comment by bartosz-maciaszek** [comment]
Happy to merge it?

**Comment by rquadling** [comment]
Let me try and break it first! :-D Will be later this evening (UK Time)

**Comment by rquadling** [comment]
Can you rebase and squash to 0.6.x-dev please?

**Comment by bartosz-maciaszek** [comment]
There you go. Should I create another pull request? This one still point to `0.5.x-dev` branch.

**Comment by rquadling** [comment]
Yes please. Squash the commits too.

**Comment by bartosz-maciaszek** [comment]
Closing. https://github.com/robmorgan/phinx/pull/875.
